src/status.rs

```diff
@@ -309,9 +427,370 @@ mod test {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            true,
+            "Follower has no new messages but leader has heartbeats and newstate"
+        )
+    }
+
+    #[test]
+    fn both_arrays_have_messages() {
+        let heartbeat_1 = get_heartbeat_msgs(Duration::minutes(0));
+        let heartbeat_2 = get_heartbeat_msgs(Duration::minutes(0));
+        let new_state = get_newstate_msg();
+        let approve_state = get_approve_state_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat_1]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(vec![heartbeat_2]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_initializing(&messages),
+            false,
+            "Both arrays have messages"
+        )
+    }
+
+    // is_disconnected()
+    #[test]
+    fn no_recent_hbs_on_both_sides() {
+        let server = SERVER_POOL.get_server();
+        let channel = get_test_channel(&server);
+        let heartbeat_1 = get_heartbeat_msgs(Duration::minutes(10));
+        let heartbeat_2 = get_heartbeat_msgs(Duration::minutes(10));
+        let new_state = get_newstate_msg();
+        let approve_state = get_approve_state_msg();
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(vec![heartbeat_1]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(vec![heartbeat_2]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_disconnected(&channel, &messages),
+            true,
+            "Both leader and follower heartbeats have no recent messages"
+        )
+    }
+
+    #[test]
+    fn no_recent_follower_hbs() {
```

Comment:
This test doesn't have any recent Heartbeat, not only on the Follower. Isn't this the same test as the one above? 
`no_recent_hbs_on_both_sides`

Reply:
It's my mistake, I'll fix it